libexec/rtld-elf/rtld_c18n.c

```diff
@@ -867,8 +868,8 @@ resolve_untrusted_stk_impl(stk_table_index index)
 uintptr_t _rtld_unw_getcontext_epilogue(uintptr_t, void **);
 struct jmp_args _rtld_unw_setcontext_epilogue(struct jmp_args, void *, void **);
 
-static void *
-unwind_cursor(void)
+void
+dl_c18n_get_trusted_stk(void **buf)
```

Comment:
Why does it need to take a `void **buf`? Just return the value and let the caller do a store if it wants to?

Reply:
This function is only called from assembly and this signature makes it a bit easier to use from assembly.

Reply:
In your current implementation it happens to only be called from assembly in places where the assembly wants to store the result into memory. That's not necessarily a justification for making the interface unusual like this.